tests/tests/NetworkModelTest.cpp

```diff
@@ -0,0 +1,485 @@
+
+  const uint8_t expected_response[] = {
+    0x00, 0x00, 0x00, 0x01,
+    0x52, 0x12, 0x34, 0x56, 0x78, 0x9a
+  };
+
+  unique_ptr<RDMResponse> response(GetResponseFromData(
+        request.get(), expected_response, arraysize(expected_response)));
+
+  int size = InvokeRDMHandler(request.get());
+  EXPECT_THAT(ArrayTuple(g_rdm_buffer, size), ResponseIs(response.get()));
+
+  // test the NR_DATA_OUT_OF_RANGE
+  interface_id = HostToNetwork(3);
+  request = BuildGetRequest(PID_INTERFACE_HARDWARE_ADDRESS_TYPE1,
+      reinterpret_cast<uint8_t*>(&interface_id), sizeof(interface_id));
+
+  response.reset(NackWithReason(request.get(), ola::rdm::NR_DATA_OUT_OF_RANGE));
+  size = InvokeRDMHandler(request.get());
+  EXPECT_THAT(ArrayTuple(g_rdm_buffer, size), ResponseIs(response.get()));
+
+  // test the NR_DATA_OUT_OF_RANGE
+  interface_id = HostToNetwork(5);
+  request = BuildGetRequest(PID_INTERFACE_HARDWARE_ADDRESS_TYPE1,
+      reinterpret_cast<uint8_t*>(&interface_id), sizeof(interface_id));
+
+  response.reset(NackWithReason(request.get(), ola::rdm::NR_DATA_OUT_OF_RANGE));
+  size = InvokeRDMHandler(request.get());
+  EXPECT_THAT(ArrayTuple(g_rdm_buffer, size), ResponseIs(response.get()));
+}
+
+TEST_F(NetworkModelTest, dhcpMode) {
+  // Get the DHCP mode for the first interface.
+  uint32_t interface_id = HostToNetwork(1);
+  unique_ptr<RDMRequest> request = BuildGetRequest(
+      PID_IPV4_DHCP_MODE,
+      reinterpret_cast<uint8_t*>(&interface_id), sizeof(interface_id));
+
+  const uint8_t expected_response[] = {
+    0x00, 0x00, 0x00, 0x01, 0x00
+  };
+
+  unique_ptr<RDMResponse> response(GetResponseFromData(
+        request.get(), expected_response, arraysize(expected_response)));
+
+  int size = InvokeRDMHandler(request.get());
+  EXPECT_THAT(ArrayTuple(g_rdm_buffer, size), ResponseIs(response.get()));
+
+  // Get the DHCP mode for the second (4th) interface.
+  interface_id = HostToNetwork(4);
+  request = BuildGetRequest(PID_IPV4_DHCP_MODE,
+      reinterpret_cast<uint8_t*>(&interface_id), sizeof(interface_id));
+
+  const uint8_t expected_response2[] = {
+    0x00, 0x00, 0x00, 0x04, 0x01
+  };
+  response.reset(GetResponseFromData(
+        request.get(), expected_response2, arraysize(expected_response2)));
+
+  size = InvokeRDMHandler(request.get());
+  EXPECT_THAT(ArrayTuple(g_rdm_buffer, size), ResponseIs(response.get()));
+
+  request = BuildGetRequest(PID_IPV4_DHCP_MODE,
+      reinterpret_cast<uint8_t*>(&interface_id), sizeof(interface_id));
+
+  // Toggle DHCP mode on the first interface.
+  const uint8_t set_param_data[] = { 0x00, 0x00, 0x00, 0x01, 0x01 };
+
+  request = BuildSetRequest(PID_IPV4_DHCP_MODE, set_param_data,
+                            arraysize(set_param_data));
+  response.reset(GetResponseFromData(request.get()));
+
+  size = InvokeRDMHandler(request.get());
+  EXPECT_THAT(ArrayTuple(g_rdm_buffer, size), ResponseIs(response.get()));
+
+  // And read it back to confirm
```

Comment:
These tests aren't supposed to be exhaustive, that sort of thing is best done in the responder tests.